vendor/github.com/containers/image/pkg/authentication/authentication.go

```diff
@@ -0,0 +1,254 @@
+package authentication
+
+import (
+	"bytes"
+	"encoding/base64"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/errors"
+)
+
+type encodedAuth struct {
+	Auth string `json:"auth"`
+}
+
+type allAuths struct {
+	Auths map[string]encodedAuth `json:"auths"`
+}
+
+// SetAuthentication stores the base64 encoding of username and password
+// in the auth file if the credentials are authenticated
+func SetAuthentication(username, password, server string) error {
+	path, err := getPathToAuth()
+	if err != nil {
+		return errors.Wrapf(err, "error getting path to auth file")
+	}
+
+	server = getFullURL(server)
+
+	isValid, err := checkAuthentication(username, password, server)
+	if err != nil {
+		return errors.Wrapf(err, "An error occured while authenticating credentials")
+	}
+	if isValid {
+		encodedCreds := encodeUserAndPass(username, password)
+		return addAuthToJSON(server, encodedCreds, path)
+	}
+	return nil
+}
+
+// GetAuthentication returns the value of the registry url stored in
+// the auth.json file
+func GetAuthentication(server string) (string, error) {
+	server = getFullURL(server)
+	path, err := getPathToAuth()
+	if err != nil {
```

Comment:
Should we also look for ~/.docker/auth.json, so if the user has done a docker login, we can succeed>

Reply:
It's your call. It can be added if you want.

Reply:
Yes I would expect it to work.  Only as secondary login.

Reply:
Added this functionality